### DISCUSSION TITLE
Relation compatible problem

### DISCUSSION THREAD

**Discussion by superiums** [mikro-orm/mikro-orm#4535] [General]
# **Is your feature request related to a problem? Please describe.**

I have an entity like this:
```ts
export class lvcourse extends BasicEntity {

  @ApiProperty({ description: 'Categeory' })
  @ManyToOne({ entity: () => dict, nullable: true, cascade: [] })
  cat: Ref<dict>;

```
dict is an object which has id, name ...

## case 1: when i query via em.find without populate option, it return records like this:
```json
{
//...
cat: 37
//...
}

```

But, while i submit an update back to db, it reports error. because in db, there's no cat. only cat_id exist.

## case 2: when i query find with {populate: 'cat'}, it return records like this:
```json
{
  //...
 cat object:{
    id: 37,
    name: ...
    // ...
 } 
}
```
It's OK.

## case 3: query with {populate : 'cat.id' }
it returns whole cat too, same to case 2.
It's not ok, i wish cat.id only.

# **Describe the solution you'd like**
 - case 1: only cat_id should be return, not cat. Only through this way,  it's compatible with update operation..
 - case 3: only cat:{ id: xxx} should be return. not whole cat object.

# **Additional context**
version 5.7.12
with nestjs 10.

**Comment by B4nan** [comment]
> case 1: only cat_id should be return, not cat. Only through this way, it's compatible with update operation..

There is no "cat_id" and "cat", its just one thing. You always get the same mapping, as otherwise it couldn't match the TS types. Note that you can have a reference in there - entity instance with just the primary key, and that will be serialized to just the PK, not an object.

> case 3: query with {populate : 'cat.id' }

You can only populate relations and lazy scalar properties, doing `populate: ['cat.id']` has no meaning, its the same as `populate: ['cat']`. Don't confuse population (for relations) and partial loading. You can do `fields: ['cat.id']`, and it should behave the way you want - v6 also improves on this. 

> case 3: only cat:{ id: xxx} should be return. not whole cat object.

The point of populating a relation is exactly getting the full relation in the response, why would you want it the other way?

You described 3 cases, all of them are rather wontfix to me, going against the ORM design principles.

With that said, you can control how entities are serialized, either via property serializer option or via explicit serialization.

https://mikro-orm.io/docs/serializing

**Reply by superiums** [reply]
thanks very much for your kind answer.
apologize for my poor knowledge of mikro.

**Reply by B4nan** [reply]
Check this guide, it's not yet complete but it already covers some basics, probably in a better-suited way than the current docs if you are completely new to this.

https://github.com/mikro-orm/guide

**Reply by superiums** [reply]
very good guid for fresher. i'll go through it. Thanks.
